src/workerd/api/sockets.c++

```diff
@@ -345,7 +349,11 @@ void Socket::handleProxyStatus(jsg::Lock& js, kj::Promise<kj::Maybe<kj::Exceptio
   // but we need the lock in our callback here.
   // TODO(cleanup): Extend awaitIo to provide the jsg::Lock in more cases.
   auto& context = IoContext::current();
-  auto result = context.awaitIo(js, kj::mv(connectResult),
+  auto result = context.awaitIo(js,
+      connectResult.catch_([](kj::Exception&& e) -> kj::Maybe<kj::Exception> {
+        LOG_ERROR_PERIODICALLY("Socket proxy disconnected abruptly", e);
+        return JSG_KJ_EXCEPTION(FAILED, Error, "connectResult raised an error");
```

Comment:
This doesn't really need to be a JSG exception, but I guess it doesn't hurt.

Reply:
I would argue that if it is to be a JSG exception, we should improve the error message so that users can understand it.

Or if it doesn't make sense to show this error to users, we should change it to a plain KJ exception.